### PR TITLE
pysdd: refactor to recursively translate into an SDD

### PR DESCRIPTION
**Pull request by tias**
may benefit from having the Boolean normalizer as transformation, once it is in master, e.g. #291

### CODE REVIEW

File: cpmpy/solvers/pysdd.py [hunk 1]
```diff
@@ -238,42 +226,108 @@ def __add__(self, cpm_expr_orig):
...
+
+        elif cpm_con.name == '!=':
+            # ~(a0 == a1)
+            equiv = self._pysdd_expr(cpm_con.args[0] == cpm_con.args[1]) 
+            return self.pysdd_manager.negate(equiv)
+            
+        elif hasattr(cpm_con, 'decompose'):
+            # for all global constraints: attempt to convert the decomposition
+            all_cons = cpm_con.decompose()  # all of these must be true, conjoin
+            return reduce(self.pysdd_manager.conjoin, [self._pysdd_expr(a) for a in all_cons])
```
Comment (JoD) [inline cpmpy/solvers/pysdd.py:321]: Will this work with the new decompose globals that gives global and local constraints?

**Comment by JoD** [comment]
Looks good, but needs to synchronize with decompose tree.

**Comment by JoD** [comment]
> may benefit from having the Boolean normalizer as transformation, once it is in master, e.g. #291

This is in the master now.

**Comment by IgnaceBleukx** [comment]
Added the new decompose-global function and simplify_boolean function to the `.transform()` of pysdd.
I think it should be ready to merge now.

**Comment by tias** [comment]
ack, ready to merge

**Pull request merged**